### PR TITLE
Add Unmatched reference

### PR DESCRIPTION
**Pull request by waveform80**
This PR fills out the HiFive Unmatched reference. Things I'm not sure about:

* Does the Live Server section apply to all the images listed (jammy, mantic, and noble)?
* Should anything be listed under Limitations?

### CODE REVIEW

File: reference/sifive-hifive-unmatched.rst [hunk 1]
```diff
@@ -2,18 +2,178 @@
...
+.. note::
+
+    U-Boot does not allow the operating system to write :term:`UEFI` variables.
+    You can do this manually using U-Boot's :command:`eficonfig` command.
+
+
+Boot source selection
+=====================
+
+The SiFive HiFive Unmatched board can boot firmware from the built-in flash
```
Comment (xypron) [inline reference/sifive-hifive-unmatched.rst:124]: In our Wiki we clear said use the 1011 setting. We never tested if 1000 would boot with our SD card.

If even the silk screen is incorrect on some boards it is getting even more confusing.

The appended picture might be more helpful to describe which setting we need.

SiFive_Unmatched_Boot_Mode_Switches

File: reference/sifive-hifive-unmatched.rst [hunk 2]
```diff
@@ -2,18 +2,178 @@
...
+#. Insert the microSD card into the board
+
+#. Set the boot source to the microSD card (see `Boot source selection`_)
+
+#. Connect the USB UART cable to the :term:`UART` header (see `UART console`_
+   and :doc:`/how-to/uart-console`)
+
+#. Power on the board
+
+#. When "Hit any key to stop autoboot" is displayed, press :kbd:`Enter`
```
Comment (xypron) [inline reference/sifive-hifive-unmatched.rst:81]: These U-Boot CLI commands are only needed if an operating system is already installed on the NVMe drive.

File: reference/sifive-hifive-unmatched.rst (hunk 1 above)
Comment (waveform80) [reply reference/sifive-hifive-unmatched.rst:116]: Pushed a couple of commits that should make this clearer

File: reference/sifive-hifive-unmatched.rst (hunk 2 above)
Comment (waveform80) [reply reference/sifive-hifive-unmatched.rst:74]: Added an update to clarify this

**Comment by waveform80** [comment]
Rebased after removal of dead-links from raspi pages

**Pull request merged**